### DISCUSSION TITLE
Override query for form and table

### DISCUSSION THREAD

**Discussion by IVoyt** [filamentphp/filament#13693] [Help] [answered]
### Package

Other

### Package Version

v3.2

### How can we help you?

Hi. Is there a way to override query separately for form and table. I know about `getEloquentQuery` method in Resource but it overrides query for both form and table of the Resource, but I need to override only for one of them or for both separately.
I tried also to override query in method `getRecord` of Action but at moment of the method call the `record` property already exists therefore there is nothing to override.

UPD:
I found I can use `modifyQueryUsing` method for table, but but there is still question how to customize query for form because `getEloquentQuery` triggered for both form and table, if I'm not mistaken. Even if it is not triggered for both methods, in my query I add leftJoin to the resource table and since base query has where clause `id = :id` without specifying resource table for id column, the id is ambiguous

**Comment by leandrocfe** [comment]
Is it a relationship? Provide more info..

**Reply by IVoyt** [reply]
no, it is join!

**Comment by IVoyt** [comment, marked as answer]
I found the solution for my case. In Resource class I can override the `resolveRecordRouteBinding` method where the ambiguous id is set.
So now I can specify `where` condition for requested resource id in `getEloquentQuery` method and inherit it's query in `resolveRecordRouteBinding`
```php
public static function getRecordId(): null|string
{
    return Route::current()->parameter('record');
}

public static function getEloquentQuery(): Builder
{
    return self::$model::query()
        ->leftJoin('some_table', 'some_table.foreign_id', '=', 'recordTable.id')
        ->where('news_id', self::getRecordId());
}

public static function resolveRecordRouteBinding(int | string $key): ?Model
{
    return self::getEloquentQuery()->first();
}
```

or I can just override `resolveRecordRouteBinding` like this
```php
public static function resolveRecordRouteBinding(int | string $key): ?Model
{
    return app(static::getModel())
        ->resolveRouteBindingQuery(static::getEloquentQuery(), $key, '<recordTable>.id')
        ->first();
}
```

**Comment by Tomodo531** [comment]
@IVoyt  You can also set the `$recordRouteKeyName`. This will have the same result.

```php
protected static ?string $recordRouteKeyName = '<recordTable>.id';
```
